files/scripts/selinux/trivalent/trivalent.te

```diff
@@ -0,0 +1,249 @@
+
+type trivalent_script_t;
+type trivalent_script_exec_t;
+domain_type(trivalent_script_t, trivalent_script_exec_t)
+application_domain(trivalent_script_t, trivalent_script_exec_t)
+
+role trivalent_script_roles types trivalent_script_t;
+
+gen_require(`
+	type audio_home_t;
+	type chrome_sandbox_home_t;
+	type config_home_t;
+	type cache_home_t;
+	type unconfined_dbusd_t;
+	type unconfined_service_t;
+	type unconfined_t;
+	type http_port_t;
+	type http_cache_port_t;
+	type howl_port_t;
+	type pki_ca_port_t;
+	type setfiles_exec_t;
+	type user_home_t;
+	type user_home_dir_t;
+	type xserver_misc_device_t;
+')
+
+# internal
+allow trivalent_t self:process { execmem getcap getsched setcap setrlimit setsched sigkill signal signull };
+allow trivalent_t self:dir { add_name create read remove_name rmdir write };
+allow trivalent_t self:file { append create execute getattr ioctl lock map open read rename unlink write };
+allow trivalent_t self:lnk_file { create read unlink };
+allow trivalent_t self:fifo_file rw_fifo_file_perms;
+allow trivalent_t self:sem create_sem_perms;
+allow trivalent_t self:netlink_kobject_uevent_socket client_stream_socket_perms;
+allow trivalent_t self:user_namespace create;
+allow trivalent_t self:unix_stream_socket connectto;
+allow trivalent_t self:cap_userns { sys_admin sys_chroot sys_ptrace };
+allow trivalent_t self:capability { dac_read_search sys_admin sys_chroot sys_ptrace };
+allow trivalent_t self:fifo_file { manage_fifo_file_perms relabelfrom relabelto };
+allow trivalent_t self:process { dyntransition transition };
+allow trivalent_t self:file manage_file_perms;
+allow trivalent_t self:dir rw_dir_perms;
+allow trivalent_t self:socket_class_set create_socket_perms;
+allow trivalent_t self:tcp_socket { accept listen };
+allow trivalent_t trivalent_exec_t:file execute_no_trans;
+allow trivalent_t chrome_sandbox_home_t:dir { add_name create read remove_name rmdir write getattr open rename search };
+allow trivalent_t chrome_sandbox_home_t:file { append create execute getattr ioctl lock map open read rename unlink write };
+allow trivalent_t chrome_sandbox_home_t:lnk_file { create read unlink };
+allow trivalent_t trivalent_home_t:dir { add_name create read remove_name rmdir write getattr open rename search };
+allow trivalent_t trivalent_home_t:file { append create execute getattr ioctl lock map open read rename unlink write };
+allow trivalent_t trivalent_home_t:lnk_file { create read unlink };
+allow trivalent_t self:netlink_route_socket nlmsg_read;
+
+# not covered by interfaces
+allow trivalent_t pki_ca_port_t:tcp_socket name_connect;
+allow trivalent_t howl_port_t:udp_socket name_bind;
+allow trivalent_t http_port_t:tcp_socket { name_connect };
+allow trivalent_t http_cache_port_t:tcp_socket { name_connect };
+allow trivalent_t tmp_t:file { create open unlink write };
+allow trivalent_t tmp_t:lnk_file { create unlink };
+allow trivalent_t tmp_t:sock_file { create unlink };
```

Comment:
on the whole of tmp_t? I don't like this - is this for something trivalent itself makes? if so, any reason we aren't doing some form of domtrans on it?